libr/core/cmd_anal.c

```diff
@@ -855,16 +855,20 @@ static void __add_vars_sdb(RCore *core, RAnalFunction *fcn) {
 
 	r_list_foreach (all_vars, iter, var) {
 		if (var->isarg) {
-			char *query = r_str_newf ("anal/types/func.%s.arg.%d=%s,%s", fcn->name, arg_count, var->type, var->name);
-			sdb_querys (core->sdb, NULL, 0, query);
-			free (query);
+			char *key = r_str_newf ("func.%s.arg.%d", fcn->name, arg_count);
+			char *value = r_str_newf ("%s,%s", var->type, var->name);
+			sdb_set (core->anal->sdb_types, key, value, 0);
+			free (key);
+			free (value);
 			arg_count++;
 		}
 	}
 	if (arg_count > 0) {
-		char *query = r_str_newf ("anal/types/func.%s.args=%d", fcn->name, arg_count);
-		sdb_querys (core->sdb, NULL, 0, query);
-		free (query);
+		char *key = r_str_newf ("func.%s.args", fcn->name);
```

Comment:
same here. no null chks and maybe use the new strbuf_initf

Reply:
Changed to strbuf.